src/app/auth/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnDestroy } from '@angular/core';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Rx';
+
+import { RegistrationCredentials , AuthActions } from '../../common/auth';
+import { getAuthState } from '../../common/store/reducers';
+
+@Component({
+  selector: 'app-sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.scss']
+})
+export class SignUpComponent  implements OnDestroy {
+  public credentials: RegistrationCredentials  = {
+    username: '',
+    password: '',
+    email: ''
+  };
+  public loading;
+  public errorMessage = '';
+  public successMessage = '';
+  private authStateSubscription: Subscription;
+
+  constructor(private router: Router, private store: Store<any>, private authActions: AuthActions) {
+    this.authStateSubscription = this.store.select(getAuthState).subscribe(({ inProgress, newUser, error }) => {
+      if (newUser) {
```

Comment:
Will we have the sign up page accessible to signed in users?
If not is `newUser` property necessary? We could just identify that signup was a success on any user data.